### PR TITLE
24 initial class structure

### LINKED ISSUE #24: Initial class structure
Determine and implement classes to allow us to handle our expected first features.  We should also consider the relationships between these different classes
- User
  - Admin
  - Student
  - Teacher
  - Other?
- Group
  - Class
  - Club
- Calendar
- Announcement
- Notification

### PR DESCRIPTION
**Pull request by rileycassidy**
Initial Class Structure first instance

### CODE REVIEW

File: classes/notifications-class.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class Notifications:
```
Comment (mckinleyevans-wrdsb) [inline classes/notifications-class.py]: Singular

File: classes/announcements-classes.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
+from datetime import datetime
+
+class announcement:
```
Comment (mckinleyevans-wrdsb) [inline classes/announcements-classes.py]: Capitalize all classnames

File: classes/announcements-classes.py
Comment (mckinleyevans-wrdsb) [inline classes/announcements-classes.py:1]: Let's just call this `announcement`

File: classes/calendar-classes.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
+class calender:
+  def __init__ (self, id)
+  self._id = id
```
Comment (mckinleyevans-wrdsb) [inline classes/calendar-classes.py]: add a `_calendar_posts` attribute

File: classes/calendar-classes.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  return self
+  
+class calender_post:
+  def __init__ (self, id, title, text, date)
+  self._title = title
+  self._text = text
+  self._date = date
+  calender.__init__(id)
```
Comment (mckinleyevans-wrdsb) [inline classes/calendar-classes.py]: remove this
We'll let a calendar track its posts, not a post track its calendar

File: classes/announcements-classes.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  def __init__ (self, id, title, text)
+  self._id = id
+  self._title = title
+  self._text = text
+  return self
+
+class sports_announcement(announcement):
+  def __init__(self, id, title, text, sport)
+  self._sport = sport
+  announcement.__init__(self, id, title, text)
```
Comment (mckinleyevans-wrdsb) [inline classes/announcements-classes.py]: Do we need to return this?

File: classes/group-class.py [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+# - Attributes
+# - name
+# - list of users
+# - Total numbers of users to be added to group
+# - Is the group active
+# - id (used from user) for staff
+class group():
+  #constructor
+  def __init__(self, groupName, userList, grouptTotal, isActive = True, staff_id)
+    self._groupName = groupName
```
Comment (mckinleyevans-wrdsb) [inline classes/group-class.py]: Use snake_case

File: classes/group-class.py [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    self._userList = userList
+    self._groupTotal = groupTotal
+    self._isActive = isActive
+    self._staff_sponsor_id = staff_id  
+    return self
+
+class classroom:
+  def __init__(self, groupName, userList, grouptTotal, isActive = True, staff_id, department)
+  self._department = department
+  group.__init__(self, groupName, userList, grouptTotal, isActive = True, staff_id)
```
Comment (mckinleyevans-wrdsb) [inline classes/group-class.py]: return this?

File: classes/users-classes.py [hunk 8]
```diff
@@ -0,0 +1,19 @@
+class user:
+ def __init__(self, id, name, pronouns)
+  self._id = id
+  self._name = name
+  self._pronouns = pronouns
+   return self
```
Comment (mckinleyevans-wrdsb) [inline classes/users-classes.py]: check the tabbing here

File: classes/announcements-classes.py [hunk 9]
```diff
@@ -0,0 +1,18 @@
+from datetime import datetime
```
Comment (mckinleyevans-wrdsb) [inline classes/announcements-classes.py]: Do we use this?

File: classes/announcements-classes.py [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+
+class announcement:
+  def __init__ (self, id, title, text)
```
Comment (mckinleyevans-wrdsb) [inline classes/announcements-classes.py]: put each input parameter on a different line
add default values for any that should have them
don't import the `id` - we'll generate that upon class initialization

Comment (mckinleyevans-wrdsb) [reply classes/announcements-classes.py]: ACTUALLY
1) Replace `id` with `uuid`
2) Generate the `uuid` inside the function, not as a parameter
https://docs.python.org/3/library/uuid.html

**Review by ghost — CHANGES_REQUESTED** [review]

File: classes/announcements-classes.py (hunk 10 above)
Comment (N0-N) [reply classes/announcements-classes.py]: Should I do the uuid for the group class as well? And can you explain how to implement the uuid?

Comment (mckinleyevans-wrdsb) [reply classes/announcements-classes.py]: Yes, replace all `id`s with `uuid`, which will all be generated inside the init of the class

https://www.uuidgenerator.net/dev-corner/python

File: classes/group-class.py [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+# - Is the group active
+# - id (used from user) for staff
+class group():
+  #constructor
+  def __init__(self, group_name, user_list, group_total, is_active = True, uuid.uuid4())
+    self._group_name = group_nameame
+    self._user_list = user_list
+    self._group_total = group_total
+    self._is_active = is_active
+    self._uuid = staff_uuid
```
Comment (mckinleyevans-wrdsb) [inline classes/group-class.py]: `self._uuid = uuid.uuid4()`
we don't have a `staff_uuid` being passed into this initialization function

File: classes/notifications-class.py (hunk 1 above)
Comment (N0-N) [reply classes/notifications-class.py]: done

File: classes/users-classes.py (hunk 8 above)
Comment (N0-N) [reply classes/users-classes.py]: done

**Review by N0-N — APPROVED** [review]
Good to go now

File: classes/calendar.py [hunk 12]
```diff
@@ -0,0 +1,9 @@
+class Calender():
+  def __init__ (self, id, calender_posts, text, date):
+    self._id = id
```
Comment (mckinleyevans-wrdsb) [inline classes/calendar.py]: `uuid`, not `id` for all classes

File: classes/group.py [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+#Class Group:
+# - Attributes
+# - name
+# - list of users
+# - Total numbers of users to be added to group
+# - Is the group active
+# - id (used from user) for staff
+class Group():
+  #constructor
+  def __init__(self, group_name, user_list, group_total, is_active = True, uuid):
```
Comment (mckinleyevans-wrdsb) [inline classes/group.py]: build the `uuid` inside the function, not as a parameter

Comment (mckinleyevans-wrdsb) [reply classes/group.py]: I see it below - you can delete it from here

File: classes/notifications.py
Comment (mckinleyevans-wrdsb) [inline classes/notifications.py:1]: rename this file to be singular

File: classes/users.py
Comment (mckinleyevans-wrdsb) [inline classes/users.py:1]: rename this file to be singular

File: classes/users.py [hunk 14]
```diff
@@ -0,0 +1,19 @@
+class User:
+ def __init__(self, id, name, pronouns):
```
Comment (mckinleyevans-wrdsb) [inline classes/users.py]: For all of these - don't pass in an `id`, create a `uuid` instead

File: classes/notifications.py [hunk 15]
```diff
@@ -0,0 +1,7 @@
+class Notification():
+  def __init__ (self, title, text, description):
+    self._title = title
```
Comment (mckinleyevans-wrdsb) [inline classes/notifications.py]: `uuid`?

File: classes/calendar.py [hunk 16]
```diff
@@ -0,0 +1,9 @@
+class Calender():
+  def __init__ (self, id, calender_posts, text, date):
```
Comment (mckinleyevans-wrdsb) [inline classes/calendar.py]: Don't pass in the `id`

File: classes/calendar.py [hunk 17]
```diff
@@ -0,0 +1,9 @@
...
+    self._id = id
+    self._calender_posts = calender_posts
+    self._text = text
+    self._date = date
+    return self
+
```
Comment (mckinleyevans-wrdsb) [inline classes/calendar.py:14]: Did we lose the `calendar_posts` class?

**Review by ghost — CHANGES_REQUESTED** [review]

File: classes/calendar.py (hunk 17 above)
Comment (N0-N) [reply classes/calendar.py:14]: yes, did you not request for it to be an attribute?

File: classes/announcement.py [hunk 18]
```diff
@@ -0,0 +1,20 @@
...
+    return self
+
+class Sports_announcement(Announcement):
+  def __init__(self, title, text, sport):
+    self._sport = sport
+    Announcement.__init__(self, title, text)
+    return self
+    
+
+class Guidence_announcement(Announcement):
```
Comment (mckinleyevans-wrdsb) [inline classes/announcement.py]: Guidance is spelled with an 'a'

File: classes/calendar.py [hunk 19]
```diff
@@ -0,0 +1,16 @@
...
+    self._uuid = uuid.uuid4()
+    self._calender_post = calender_post
+    self._event = event
+    return self
+
+class Calendar_post(Calendar):
+  def __init__(self, event, text, date):
+    self._text = text
+    self._date = date
+    Calendar.__init__(self, Calendar_post, event)
```
Comment (mckinleyevans-wrdsb) [inline classes/calendar.py]: Scrap this.  We'll create posts separately, then add them all into one calendar

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**